### PR TITLE
Moving subquery execution to a worker thread (continuation of #83)

### PR DESCRIPTION
**Pull request by arsen-es**
*Issue #, if available:* N/A

*Description of changes:* This is to replace PR #83, since the code on that one was a little outdated. Merged from master, resolved conflicts, also addressed the comments in #83. Verified on local cluster that the subquery execution does not fail.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by arsen-es** [comment]
Note, that the code structuring is not ideal, I just wanted to make a single file change so that when we have the better fix, it's easier to compare/undo the changes.

File: src/main/java/com/amazon/opendistroforelasticsearch/sql/plugin/RestSqlAction.java [hunk 1]
```diff
@@ -138,7 +142,55 @@ protected RestChannelConsumer prepareRequest(RestRequest request, NodeClient cli
...
+                if (request.hasParam(paramName)) {
+                    additionalParams.put(paramName, request.param(paramName));
+                }
+            }
+            restExecutor.execute(client, additionalParams, queryAction, channel);
+        }
+    }
+
+    private static boolean isExplainRequest(final RestRequest request) {
+
```
Comment (galkk) [inline src/main/java/com/amazon/opendistroforelasticsearch/sql/plugin/RestSqlAction.java]: remove empty line

**Comment by galkk** [comment]
Good job

File: src/main/java/com/amazon/opendistroforelasticsearch/sql/plugin/RestSqlAction.java (hunk 1 above)
Comment (arsen-es) [reply src/main/java/com/amazon/opendistroforelasticsearch/sql/plugin/RestSqlAction.java]: done

**Pull request merged**